### PR TITLE
Return scalar losses instead of per-sample means

### PR DESCRIPTION
**Pull request by Rocketknight1**
This updates the TF XLA-compatible losses to return scalars instead of per-sample means. As @ydshieh pointed out, per-sample means give too much weight to samples with fewer masked positions. The new approach should match PyTorch losses exactly (up to floating-point error).

TODO:
- [x] Update expected sizes in tests

### CODE REVIEW

**Review by ydshieh — APPROVED** [review]
LGTM, thank you @Rocketknight1!

- I didn't see the change in RAG model (which you modified in the last PR and we have some test failures)
  - You want to work on  this in a separate PR?
- Could you run (some of) the current (related) failed tests before merge?

**Comment by Rocketknight1** [comment]
@ydshieh will investigate RAG and make sure tests pass before merging!

**Comment by Rocketknight1** [comment]
@ydshieh I reverted the RAG loss function to the pre-XLA version, so hopefully those tests pass now. All other tests are passing! Do you think there's anything else I'm missing before I merge?

**Review by gante — APPROVED** [review]
LGTM!

Do we have some tests for TF<>PT loss equivalence? If not, would it be hard to set them?

**Comment by Rocketknight1** [comment]
@gante We do, but it [takes the mean of the TF loss in order to make the two losses comparable](https://github.com/huggingface/transformers/blob/main/tests/test_modeling_tf_common.py#L560-L567).

**Comment by ydshieh** [comment]
> Do you think there's anything else I'm missing before I merge?

Nothing I can think of as the tests pass now 🎉 . Thank you!

File: src/transformers/modeling_tf_utils.py [hunk 1]
```diff
@@ -206,11 +206,9 @@ def hf_compute_loss(self, labels, logits):
         unmasked_loss = loss_fn(tf.nn.relu(labels), logits)
         # make sure only labels that are not equal to -100 affect the loss
         loss_mask = tf.cast(labels != -100, dtype=unmasked_loss.dtype)
-        # Avoid division by zero later
-        loss_denominator = tf.math.maximum(tf.cast(1, loss_mask.dtype), tf.reduce_sum(loss_mask, axis=1))
         masked_loss = unmasked_loss * loss_mask
-        reduced_masked_loss = tf.reduce_sum(masked_loss, axis=1) / loss_denominator
-        return reduced_masked_loss
+        reduced_masked_loss = tf.reduce_sum(masked_loss) / tf.reduce_sum(loss_mask)
+        return tf.reshape(reduced_masked_loss, (1,))
```
[the remark this answers is not in the document]
Comment (ydshieh) [reply src/transformers/modeling_tf_utils.py:211]: OK, I respect Keras!
